### PR TITLE
Disable v1beta1 audit types until we have made all incompatible changes.

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go [hunk 1]
```diff
@@ -88,24 +89,30 @@ var (
...
-		panic(fmt.Sprintf("failed to enable version %v", groupVersions))
+	allGVs := []schema.GroupVersion{auditv1alpha1.SchemeGroupVersion, auditv1beta1.SchemeGroupVersion}
+	registry.RegisterVersions(allGVs)
+	numEnabled := 0
+	for _, gv := range allGVs {
+		if err := registry.EnableVersions(gv); err == nil {
+			numEnabled++
+		}
+	}
+	if numEnabled == 0 {
```
Comment (crassirostris) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: Why this change? I mean only panic if none of the gv's were enabled, not if failed to enable any

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go [hunk 2]
```diff
@@ -70,12 +71,12 @@ const pluginName = "webhook"
 // NewBackend returns an audit backend that sends events over HTTP to an external service.
 // The mode indicates the caching behavior of the webhook. Either blocking (ModeBlocking)
 // or buffered with batch POSTs (ModeBatch).
-func NewBackend(kubeConfigFile string, mode string) (audit.Backend, error) {
+func NewBackend(kubeConfigFile string, mode string, groupVersions []schema.GroupVersion) (audit.Backend, error) {
```
Comment (crassirostris) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:74]: Nit: switch to variadic maybe? I imagine for a webhook at least only one version is required

**Review by crassirostris — COMMENTED** [review]
LGTM

Maybe split beta tests to a separate file to avoid doing so when moving to v1?

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:74]: the versions are not the primary params of a backend IMO. Variadic would imply that.

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: if you restrict the enabled apis, at least one should be enabled, but maybe not all of them. But this is only important when we support some kind of negotiation.

Though now you ask I wonder: will we crash if alpha features are disabled?

Comment (crassirostris) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: Is it possible to do this step later, when the preferred version is known?

Comment (sttts) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: Yes, am fine with that. @CaoShuFeng can you remove this change?

Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: OK.
Your are so fast. I am still try to understand this patch.

Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go:104]: Removed.
